### PR TITLE
Handle seconds in `zip:dos_date_time_from_datetime/1`

### LINKED ISSUE #7412: zip:create/2,3 does not accept a valid file:file_info record
**Describe the bug**
`zip:create/2,3` does not accept a valid `file:file_info` record.

**To Reproduce**
In erl:
```
Eshell V13.2.2  (abort with ^G)
1> rr(code:lib_dir(kernel, include) ++ "/file.hrl").
[file_descriptor,file_info]
2> zip:create("archive.zip", [{"file.txt", <<>>, #file_info{type=regular, mtime=0}}], [memory]).
{error,{'EXIT',{function_clause,[{zip,dos_date_time_from_datetime,
                                      [0],
                                      []},
                                 {zip,local_file_header_from_info_method_name,5,
                                      [{file,"zip.erl"},{line,1131}]},
                                 {zip,put_z_files,6,[{file,"zip.erl"},{line,874}]},
                                 {zip,do_zip,3,[{file,"zip.erl"},{line,466}]},
                                 {zip,zip,3,[{file,"zip.erl"},{line,455}]},
                                 {erl_eval,do_apply,7,[{file,"erl_eval.erl"},{line,748}]},
                                 {shell,exprs,7,[{file,"shell.erl"},{line,691}]},
                                 {shell,eval_exprs,7,[{file,"shell.erl"},{line,647}]}]}}}
```

**Expected behavior**
The file modification time in the archive is set to the Unix epoch. (This won't actually happen because I've omitted the rest of the file_info for brevity.)

**Affected versions**
At least 25.3.2

**Additional context**
I want to create a reproducible archive and while it's doable with the current API it's not that concise. It might be worth adding an option for this.

### PR DESCRIPTION
**Pull request by sirikid**
Fixes #7412

### CODE REVIEW

File: lib/stdlib/src/zip.erl [hunk 1]
```diff
@@ -1510,7 +1510,7 @@ get_z_file(In0, Z, Input, Output, OpO, FB,
 					   CompSize, Input, Output, OpO, Z),
 			    In5 = skip_z_data_descriptor(GPFlag, Input, In4),
 			    %% TODO This should be fixed some day:
-			    %% In5 = Input({set_file_info, FileName, 
+			    %% In5 = Input({set_file_info, FileName,
```
Comment (jhogberg) [inline lib/stdlib/src/zip.erl:1513]: Nitpick: please skip this unrelated whitespace change.

**Review by jhogberg — CHANGES_REQUESTED** [review]
Thanks for the PR! All that's lacking is a test case, can you add one to `zip_SUITE.erl`? :-)

File: lib/stdlib/src/zip.erl (hunk 1 above)
Comment (sirikid) [reply lib/stdlib/src/zip.erl:1513]: Sure

**Review by jhogberg — APPROVED** [review]

**Comment by jhogberg** [comment]
Merged, thanks again for the PR! :)

**Pull request merged**